pkg/controller/drain/drain_controller.go

```diff
@@ -318,8 +325,19 @@ func (ctrl *Controller) syncNode(key string) error {
 		// Attempt drain
 		ctrl.logNode(node, "initiating drain")
 		if err := drain.RunNodeDrain(drainer, node.Name); err != nil {
-			ctrl.logNode(node, "Drain failed. Waiting 1 minute then retrying. Error message from drain: %v", err)
-			ctrl.enqueueAfter(node, drainRequeueDelay)
+			// To mimic our old daemon logic, we should probably have a more nuanced backoff.
+			// However since the controller is processing all drains, it is less deterministic how soon the next drain will retry,
+			// Anywhere between instant (if a node change happened) or up to hours (if there are many nodes competing for resources)
+			// For now, let's say if a node has been trying for a set amount of time, we make it less prioritized.
+			if duration > drainRequeueFailingThreshold {
+				ctrl.logNode(node, "Drain failed. Drain has been failing for more than %v minutes. Waiting %v minutes then retrying. "+
+					"Error message from drain: %v", drainRequeueFailingThreshold.Minutes(), drainRequeueFailingDelay.Minutes(), err)
```

Comment:
Instead of `drainRequeueFailingThreshold.Minutes()', shouldn't we print `duration` as it will give more precise info?

Reply:
The duration is printed above via `glog.Infof("Previous node drain found. Drain has been going on for %v hours", duration.Hours())`, in theory that line should be right above this one

Reply:
If the threshold is confusing/unnecessary to print, I can also remove that

Reply:
No strong feeling, let's keep it.